source/parser.hera

```diff
@@ -3110,10 +3109,7 @@ BinaryOpSymbol
     }
   "<<"
   "«" -> "<<"
-  # NOTE: Avoid matching JSX opening tag by requiring non-identifier character
-  # (e.g. whitespace) after "<".  This does forbid 1<2 or x<y.
-  /<(?!\p{ID_Start}|[_$])/ ->
-    return "<"
+  "<"
```

Comment:
Kind of surprised this works! But must be an old way of avoiding ambiguity.

Reply:
It originally caused some trouble with `class X extends A<B>` but switching to enforcing `ExtendsTarget` as a `LeftHandSideExpression` fixes it by requiring parens around expressions.